### ISSUE TITLE
Query BLOB SUB_TYPE TEXT pastes the code "function(callback) { // ..." into the field

### ISSUE THREAD

**Issue by sh0men** [hgourvest/node-firebird#206]
BLOB SUB_TYPE TEXT field contains a character ( „ and ” ) that seems to be the issue, therefore I've tried to replace them. It works fine with other tools but not with node-firebird.

Database CHARSET WIN-1250.

Following scenarios have been tried:

1. Scenario
**Code:**

var statement="SELECT FIRST 1000 SKIP 0 ID, cast(OTHERBLOBFIELD as varchar(1000) character set WIN1250) OTHERBLOBFIELD , replace(replace(LONGTEXTFIELD, '„', '"'), '”', '"')  LONGTEXTFIELD FROM TABLE1 where ID like '170' order by ID ;"

... 
           Firebird.attach(options, function(err, db) {
                try {
                    db.query(statement, function(error, result) {
                        if (error)
                            reject(error);
                        resolve(result);
                        db.detach();
                    });
                }  catch ( err ) {
                    reject(err);
                }
            });
...

**Column LONGTEXTFIELD received:** "function(callback) {
        // callback(err, buffer, name);
        statement.connection.startTransaction(ISOLATION_READ_UNCOMMITTED, function(err, transaction) {....."

2. Scenario
**Code:**

var statement="SELECT FIRST 1000 SKIP 0 ID, cast(OTHERBLOBFIELD as varchar(1000) character set WIN1250) OTHERBLOBFIELD , CAST(replace(replace(LONGTEXTFIELD, '„', '"'), '”', '"') as varchar(1000)) LONGTEXTFIELD FROM TABLE1 where ID like '170' order by ID ;"

... 
           Firebird.attach(options, function(err, db) {
                try {
                    db.query(statement, function(error, result) {
                        if (error)
                            reject(error);
                        resolve(result);
                        db.detach();
                    });
                }  catch ( err ) {
                    reject(err);
                }
            });
...

**OUTPUT** On those rows where the problematic characters can be found the application stops working (no output, no error)...

I've tried many things, the characters generate issue also in other tools, iSQL throws in some circumstances Malformed String error, too. But it throws something. With node-firebird I cannot even get a real output even with bad coding.

**Comment by goulu** [comment]
check the example on https://www.npmjs.com/package/node-firebird : the blob can be read asynchronously. What you see is the callback function that reads it.

Now my problem is to read it synchronously ...

**Comment by mariuz** [comment]
Two separate things were happening here:

1. **`function(callback) { ... }` appearing in the field**: blob columns come back as async reader *functions* by default; stringifying the row turns them into source text. `blobAsText: true` returns SUB_TYPE TEXT blobs as plain strings — see the [FAQ](https://github.com/hgourvest/node-firebird#faq).
2. **WIN1250 characters (`„ ”`)**: Node has no built-in WIN1250 codec, so the driver cannot transcode that charset natively. The FAQ now documents the working recipe (`encoding: 'NONE'` + iconv-lite in a `typeCast` hook): [Text comes back as ������ with a WIN125x database](https://github.com/hgourvest/node-firebird#faq).

Closing as answered.
